.github/workflows/deploy.yml

```diff
@@ -39,10 +39,19 @@ jobs:
       if: github.ref == 'refs/heads/beta'
       run: echo "BETA=true" >> $GITHUB_ENV
 
-    - name: Use scripts and templates from master branch
+    - name: Use deploy script from master branch
       run: |
         git fetch --no-tags --prune --depth=1 origin master
-        git checkout origin/master -- .github/deploy.sh util/gh-pages/ util/*.py
+        git checkout origin/master -- .github/deploy.sh
+
+    - name: cargo update
+      run: cargo update
+
+    - name: Cache
+      uses: Swatinem/rust-cache@v1
```

Comment:
Yes, will do. This is **really** WIP and I haven't tested anything yet. This PR was more for writing down my notes rather than the finished CI changes.